### PR TITLE
Suheyb - Week2

### PR DESCRIPTION
**Pull request by suh3yb**

### CODE REVIEW

File: week2/homework/src/actions-bonus.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const fs = require('fs');
+
+const jsonFilePath = 'list.json';
+
+function readAndParse() {
```
Comment (gajduk) [inline week2/homework/src/actions-bonus.js]: use of a separate function for loading and parsing 💯

File: week2/homework/src/actions-bonus.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  return new Promise((resolve, reject) => {
+    fs.readFile(jsonFilePath, 'utf8', (error, listJson) => {
+      if (error) return reject(error);
+      const listObj = JSON.parse(listJson);
```
Comment (gajduk) [inline week2/homework/src/actions-bonus.js]: use of JSON 👍

File: week2/homework/src/actions-bonus.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        resolve(list.trim());
+      })
+      .catch(error => reject(error));
+  });
+};
+
+const add = toDo => {
+  return new Promise((resolve, reject) => {
+    if (!toDo) {
+      return reject(new Error('Please provide a to-do.'));
```
Comment (gajduk) [inline week2/homework/src/actions-bonus.js]: Nice use of reject. Throwing a custom error 👍

File: week2/homework/src/index-bonus.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const program = require('commander');
```
Comment (gajduk) [inline week2/homework/src/index-bonus.js]: use of comander 👍

**Comment by suh3yb** [comment]
Thank you so much for your time and feedback, @gajduk. I am glad you like my work. :smiley:
Best regards.

**Pull request closed without merging**